lib/karafka/setup/config.rb

```diff
@@ -285,6 +285,8 @@ class Config
           setting :expansions_selector, default: Processing::ExpansionsSelector.new
           # option [Class] executor class
           setting :executor_class, default: Processing::Executor
+          # option external_worker_executor [proc] callable object that will be used to wrap execution of a job
+          setting :external_worker_executor, default: proc { |&block| block.call }
```

Comment:
not sure if I would call it `executor` per se - maybe more of a wrapper and I would set it to `nil` for defaults and not use it if not needed. That way we don't have to have a second nesting in this critical (but I can do it myself)